src/Transports.AspNetCore/WebSockets/IWebSocketConnection.cs

```diff
@@ -0,0 +1,50 @@
+using GraphQL.Transport;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Server.Transports.AspNetCore.WebSockets;
+
+/// <summary>
+/// Represents a WebSocket connection, dispatching received messages over
+/// the connection to the specified <see cref="IOperationMessageProcessor"/>,
+/// and sending requested messages out the connection when requested.
+/// <br/><br/>
+/// <see cref="ExecuteAsync(IOperationMessageProcessor)"/> may be only called once.
+/// All members must be thread-safe.
+/// </summary>
+public interface IWebSocketConnection : IDisposable
+{
+    /// <summary>
+    /// Listens to incoming messages over the WebSocket connection,
+    /// dispatching the messages to the specified <paramref name="operationMessageProcessor"/>.
+    /// Returns or throws <see cref="OperationCanceledException"/> when the WebSocket connection is closed.
+    /// </summary>
+    Task ExecuteAsync(IOperationMessageProcessor operationMessageProcessor);
+
+    /// <summary>
+    /// Sends a message to the client.
+    /// </summary>
+    Task SendMessageAsync(OperationMessage message);
+
+    /// <summary>
+    /// Performs a graceful shutdown of the WebSocket connection with event ID 1000.
```

Comment:
> ID 1000

Required by protocol spec?

Reply:
The WebSocket spec, yes. See https://docs.microsoft.com/en-us/dotnet/api/system.net.websockets.websocketclosestatus?view=net-6.0

I did not use the enum since the number is arbitrary and the GraphQL over WebSocket spec defines other event id numbers to use for specific situations which are not in the enum.

That may be a good use of an extension method - an “overload” with the status enum instead of the int.